library-core/src/main/java/com/mikepenz/fastadapter/items/SimpleItemList.java

```diff
@@ -0,0 +1,174 @@
+package com.mikepenz.fastadapter.items;
+
+import com.mikepenz.fastadapter.IItem;
+import com.mikepenz.fastadapter.adapters.IItemList;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+
+public class SimpleItemList<Item extends IItem> extends IItemList<Item> {
+  private final List<Item> list;
+
+  public SimpleItemList() {
+    this(-1);
+  }
+  
+  public SimpleItemList(int initialSize) {
+    list = initialSize < 0 ? new ArrayList<Item>() : new ArrayList<Item>(initialSize);
+  }
+  
+  @Override
+  public int size() {
+    return list.size();
+  }
+
+  @Override
+  public Item get(int position) {
+    return list.get(position);
+  }
+
+  @Override
+  public List<Item> getAll() {
+    return new ArrayList<>(list);

```

Comment:
better give the correct reference back even when thats not possible in the sorted list.

Reply:
See my comment above about exposing internal state